### ISSUE TITLE
No detailed description of config is given.

### ISSUE THREAD

**Issue by udemegane** [nv-tlabs/datasetGAN_release#10]
Hi,
To run datasetgan, i need not only the model represented by .pt but also the npy binaries for average_latent and annotation_image_latent. However, the readme only has a configuration that works based on four pre-prepared config files.
I understand that I need to write my own config and prepare the necessary data, but there is no explanation for that.
Or is this code just a demo and not supposed to work with any data sets?
Thank you.

**Comment by arieling** [comment]
Hey once you load the average_latent and annotation_image_latent. you can follow the data format to prepare your data. Should be straightforward.

**Comment by PoissonChasseur** [comment]
As suggested, I also looked at the format of these latent space files and I get this:
```
Folder cat_16 
	 File avg_latent_stylegan1.npy 
		 shape = (18, 512) 
		 nb element = 9216 
		 nb unique value = 512 
		 [Min, Max] = [-0.224304, 0.364038] 
		 Mean ± std = 0.002004 ± 0.055711 

	 File latent_stylegan1.npy 
		 shape = (30, 512) 
		 nb element = 15360 
		 nb unique value = 15360 
		 [Min, Max] = [-4.295391, 4.562115] 
		 Mean ± std = 0.010828 ± 0.999092 


Folder car_20 
	 File avg_latent_stylegan1.npy 
		 shape = (18, 512) 
		 nb element = 9216 
		 nb unique value = 512 
		 [Min, Max] = [-0.060179, 0.156538] 
		 Mean ± std = 0.003713 ± 0.022411 

	 File latent_stylegan1.npy 
		 shape = (33, 512) 
		 nb element = 16896 
		 nb unique value = 16896 
		 [Min, Max] = [-3.715559, 3.471549] 
		 Mean ± std = -0.000713 ± 0.997413 


Folder face_34 
	 File avg_latent_stylegan1.npy 
		 shape = (18, 512) 
		 nb element = 9216 
		 nb unique value = 512 
		 [Min, Max] = [-0.124208, 0.286845] 
		 Mean ± std = 0.019030 ± 0.055279 

	 File latent_stylegan1.npy 
		 shape = (16, 512) 
		 nb element = 8192 
		 nb unique value = 8192 
		 [Min, Max] = [-3.766154, 3.863906] 
		 Mean ± std = 0.011006 ± 1.009952  
```
Although 512 is probably the size of the StyleGAN's W latent space, I really don't see the relationship between these dimensions and the number of channels for "dim" (in the ".json" files) and how we should get these latent space files.

**Comment by arieling** [comment]
avg_latent_stylegan1 is average of w latent space.
latent_stylegan1 is w to construct training images.  (16, 512) means there are 16 training images.

**Comment by PoissonChasseur** [comment]
By reading the code further, I could indeed see that "**latent_stylegan1.npy**" was related to the latent codes Z (or W - according to "annotation_data_from_w") of the training images which must have been generated by the StyleGAN. These must have been pre-generated and their label masks must also have been made on them.

However, "**avg_latent_stylegan1**" seems to be used only before loading pre-trained weights via the "stylegan_chekpoint" and in the link for weight transfer from [Tensorflow -> PyTorch] (issue # 1) , they don't seem to use it.

**Due to this, I'm not sure how to get the "avg_latent_stylegan1".** Is this from the pre-trained StyleGAN or just an average of the latent space of the "latent_stylegan1.npy" (or something else) ? 

Also, is it possible to pre-calculate the input number of the first FC of the "pixel_classifier" (dim [-1]) ?

**Comment by arieling** [comment]
avg_latent_stylegan1 is the mean of 8000 randomly sampled w.

**Comment by PoissonChasseur** [comment]
Ah ok, thank you very much for that. I imagine the reason there are only 512 unique values in the shape matrices (18, 512) is that the values of the mean vector of shape (1, 512) are duplicated along the 0 axis.

And for the value of **dim [-1]** (the number of channels in the "dim" parameter of the configuration files), **how can we get it without first running the code with an invalid value?**
The fact that this value changes between different datasets (even those with images of the same size) seems a bit strange from my perspective.

**Comment by arieling** [comment]
ah that's because the different category has different resolution such that the number of layers in stylegan are different. You can calculate the dim[-1] by counting stylegan features. However, I would just run the code and error message will tell you the number dim[-1].

**Comment by 1goodone** [comment]
I have the same problem, so how do I generate average_latent and annotation_image_latent
thankyou

**Comment by arieling** [comment]
@1goodone please see the comments above

**Comment by 1goodone** [comment]
@arieling thankyou

**Comment by ericchansen** [comment]
@PoissonChasseur, do you have examples that describe how you generated these files?

Alternatively, @arieling, can you please provide more docs on how to generate these files?
